include/pybind11/detail/argument_vector.h

```diff
@@ -192,7 +193,7 @@ struct argument_vector {
     }
 
 private:
-    using repr_type = inline_array_or_vector<handle, N>;
+    using repr_type = inline_array_or_vector<T, InlineSize>;
     repr_type m_repr;
 
     PYBIND11_NOINLINE void move_to_heap_vector_with_reserved_size(std::size_t reserved_size) {
```

Comment:
As this is written except for the assertion, this has strong noexcept guarnatees vs implace move assignment